### ISSUE TITLE
BulkheadException: when using @Retry with the rest-client

### ISSUE THREAD

**Issue by loicmathieu** [quarkusio/quarkus#6073]
**Describe the bug**
When using `@Retry` with the Quarkus REST Client (standard, not async), I have some exception thrown when doing a load test.

```
BulkheadException: could not acquire a semaphore for execution and no fallback available
```

After some discussion on Zulip: https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Fault.20tolerance.20.22could.20not.20acquire.20a.20semaphore.22.20error it appears that the Fault Tolerance extension uses Hystix that internaly use semaphore as it's concurency model.

A workaround is to increase the number of semaphore using the following system property: `hystrix.command.default.execution.isolation.semaphore.maxConcurrentRequests`.

As a call to a rest client is an I/O operation, setting the number of semaphore to the number of Vert.X worker thread seems a good choice for me, and experimentation (on my load test) shows that doing so fixes the issue.

**Expected behavior**
No exception throws on fault tolerance code.

**To Reproduce**
1. Create a Rest Client that connect to an external service
2. Add a @Retry annotation
3. Put loads on your application so that you have more worker threads doing I/O operation on the rest client than the default number of Hystrix semaphore (10).

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Linux 5.3.0-24-generic #26-Ubuntu SMP Thu Nov 14 01:33:18 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
 - Output of `java -version`: openjdk version "11.0.5" 2019-10-15
 - Quarkus version or git rev: 1.0.1.Final

**Additional context**
(Add any other context about the problem here.)

**Comment by loicmathieu** [comment]
I propose to automatically configure the number of Hystrix semaphore to the numver of Vert.x worker thread (20 per default) inside the fault tolerance extension.

This should be OK for fault tolerance features around I/O operation (rest client, database) but not needed for fault tolerance on "memory operation" so maybe it's not a good idea.

Feedback are welcome on this topic.

**Comment by loicmathieu** [comment]
@michalszynkiewicz @mkouba now that #6300 has been merged, `BulkheadException` cannot occurs as FT is not more implemented using Hystrix right ?

But it still uses Semaphore, is the number of semaphore aligned with the number of worker thread so the root cause of this issue with not occurs after #6300 ?

**Comment by michalszynkiewicz** [comment]
@loicmathieu sorry, I don't fully understand. Are you saying the issue is no longer there with #6300 merged and asking why it might be?

The new implementation does not use Hystrix, yes. But it still has a limit on the maximum number of threads used by the fault tolerance mechanisms. It defaults to 100 and can be modified in application.properties, see https://github.com/smallrye/smallrye-fault-tolerance/#configuration
Please note that this limit does not impact the pools used for bulkheads.

**Comment by loicmathieu** [comment]
@michalszynkiewicz yes, you seems to totaly understand ;)

This issue was that using @Retry ends up with BulheadException because Hystix uses Bulhead for running commands. The root cause was that Hystrix uses 10 semaphore for concurrency and that Quarkus uses 20 worker threads by default so you could ends up using all semaphores and generating this exception.

Now that FT didn't use Hystrix anymore and that the concurency limit is greater that the default number of worker thread, this issue can be closed.
